src/problem2.rb

```diff
@@ -0,0 +1,46 @@
+class Problem2
+  def calculateMaximum(values) #theta(n/2) -> O(n)
+    response = [nil,nil]
+    values.each_slice(2) do |first,second|
+      setMaxNumbers response,first,second
+    end
+    switchMaxValues response
+    return response
+  end
+
+  def setMaxNumbers(maxValues, firstValue, secondValue)
+    if maxValues[0] == nil && maxValues[1] == nil
+      maxValues[0] = firstValue
+      maxValues[1] = secondValue.nil? ? firstValue : secondValue
+      return
+    end
+    compareValues maxValues,firstValue
+    if secondValue != nil
+      compareValues maxValues,secondValue
+    end
+  end
+
+  def compareValues(maxValues, newValue)
+
+    if maxValues[0] <= newValue || maxValues[1] <= newValue
+      replaceMaxNumbers maxValues, newValue
```

Comment:
`compareValues` does much more than it advertises (would expect it to return a comparison result aka one of [-1, 0, 1]). In this case it mutates one of the parameters.

Reply:
evaluateValues or a similar name would be more clear maybe.